### PR TITLE
Migrate price reference resolver

### PR DESCRIPTION
**Pull request by lojzatran**
#### Summary

Migrate price reference resolver and its tests to sdk v2. This is part of the preparation for migrating VariantReferenceResolver later.

### CODE REVIEW

File: src/main/java/com/commercetools/sync/sdk2/inventories/helpers/InventoryEntryIdentifier.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+import com.commercetools.api.models.channel.ChannelResourceIdentifier;
+import com.commercetools.api.models.inventory.InventoryEntry;
+import com.commercetools.api.models.inventory.InventoryEntryDraft;
+import java.util.Objects;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+/**
+ * This class provides as a container of the unique identifier of an {@link
+ * io.sphere.sdk.inventory.InventoryEntry} for the sync which is a combination of the SKU and the
```
Comment (salander85) [inline src/main/java/com/commercetools/sync/sdk2/inventories/helpers/InventoryEntryIdentifier.java:15]: Please check the links in comments and update to sdk2.

File: src/main/java/com/commercetools/sync/sdk2/products/helpers/PriceReferenceResolver.java [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+      @Nonnull final TypeService typeService,
+      @Nonnull final ChannelService channelService,
+      @Nonnull final CustomerGroupService customerGroupService) {
+    super(options, typeService);
+    this.channelService = channelService;
+    this.customerGroupService = customerGroupService;
+  }
+
+  /**
+   * Given a {@link io.sphere.sdk.products.PriceDraft} this method attempts to resolve the custom
```
Comment (salander85) [inline src/main/java/com/commercetools/sync/sdk2/products/helpers/PriceReferenceResolver.java:66]:
```suggestion
   * Given a {@link com.commercetools.api.models.common.PriceDraft} this method attempts to resolve the custom
```

File: src/main/java/com/commercetools/sync/sdk2/products/helpers/PriceReferenceResolver.java [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+                                    } else {
+                                      result.complete(draftWithCreatedChannel);
+                                    }
+                                  });
+                          return result;
+                        }));
+  }
+
+  /**
+   * Helper method that creates a new {@link io.sphere.sdk.channels.Channel} on the CTP project with
```
Comment (salander85) [inline src/main/java/com/commercetools/sync/sdk2/products/helpers/PriceReferenceResolver.java:184]:
```suggestion
   * Helper method that creates a new {@link com.commercetools.api.models.channel.Channel} on the CTP project with
```

File: src/main/java/com/commercetools/sync/sdk2/products/helpers/PriceReferenceResolver.java [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+              });
+    } else {
+      final ReferenceResolutionException referenceResolutionException =
+          new ReferenceResolutionException(format(CHANNEL_DOES_NOT_EXIST, channelKey));
+      return exceptionallyCompletedFuture(referenceResolutionException);
+    }
+  }
+
+  /**
+   * Given a {@link io.sphere.sdk.products.PriceDraftBuilder} this method attempts to resolve the
```
Comment (salander85) [inline src/main/java/com/commercetools/sync/sdk2/products/helpers/PriceReferenceResolver.java:230]:
```suggestion
   * Given a {@link com.commercetools.api.models.common.PriceDraftBuilder} this method attempts to resolve the
```

File: src/main/java/com/commercetools/sync/sdk2/products/helpers/PriceReferenceResolver.java [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+      @Nonnull final String channelKey, @Nonnull final PriceDraftBuilder draftBuilder) {
+
+    if (options.shouldEnsurePriceChannels()) {
+      return channelService
+          .createAndCacheChannel(channelKey)
+          .thenCompose(
+              createdChannelOptional -> {
+                if (createdChannelOptional.isPresent()) {
+                  return completedFuture(
+                      draftBuilder.channel(createdChannelOptional.get().toResourceIdentifier()));
```
Comment (salander85) [inline src/main/java/com/commercetools/sync/sdk2/products/helpers/PriceReferenceResolver.java:215]: Note that `toResourceIdentifier()` doesn't set the key. Might be better to use `ChannelResourceIdentifierBuilder.of().key(createdChannelOptional.get().getKey()).build();` instead.

File: src/test/java/com/commercetools/sync/sdk2/inventories/InventorySyncMockUtils.java [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import javax.annotation.Nonnull;
+
+public class InventorySyncMockUtils {
+
+  /**
+   * Returns mock {@link io.sphere.sdk.channels.Channel} instance. Returned instance represents
```
Comment (salander85) [inline src/test/java/com/commercetools/sync/sdk2/inventories/InventorySyncMockUtils.java:29]: Please change the links in the comments.

**Review by salander85 — CHANGES_REQUESTED** [review]

File: src/main/java/com/commercetools/sync/sdk2/inventories/helpers/InventoryEntryIdentifier.java (hunk 1 above)
Comment (lojzatran) [reply src/main/java/com/commercetools/sync/sdk2/inventories/helpers/InventoryEntryIdentifier.java]: https://github.com/commercetools/commercetools-sync-java/pull/931/commits/37c82ff9880900242f304c60a89bf06e4ede69fd

File: src/main/java/com/commercetools/sync/sdk2/products/helpers/PriceReferenceResolver.java (hunk 2 above)
Comment (lojzatran) [reply src/main/java/com/commercetools/sync/sdk2/products/helpers/PriceReferenceResolver.java]: https://github.com/commercetools/commercetools-sync-java/pull/931/commits/37c82ff9880900242f304c60a89bf06e4ede69fd

File: src/main/java/com/commercetools/sync/sdk2/products/helpers/PriceReferenceResolver.java [hunk 7]
```diff
@@ -0,0 +1,296 @@
...
+        PriceDraftBuilder::getCustom,
+        PriceDraftBuilder::custom,
+        format(
+            FAILED_TO_RESOLVE_CUSTOM_TYPE,
+            draftBuilder.getCountry(),
+            draftBuilder.getValue().toMonetaryAmount()));
+  }
+
+  /**
+   * Given a {@link io.sphere.sdk.products.PriceDraftBuilder} this method attempts to resolve the
```
[the remark this answers is not in the document]
Comment (lojzatran) [reply src/main/java/com/commercetools/sync/sdk2/products/helpers/PriceReferenceResolver.java]: https://github.com/commercetools/commercetools-sync-java/pull/931/commits/37c82ff9880900242f304c60a89bf06e4ede69fd

File: src/main/java/com/commercetools/sync/sdk2/products/helpers/PriceReferenceResolver.java (hunk 3 above)
Comment (lojzatran) [reply src/main/java/com/commercetools/sync/sdk2/products/helpers/PriceReferenceResolver.java]: https://github.com/commercetools/commercetools-sync-java/pull/931/commits/37c82ff9880900242f304c60a89bf06e4ede69fd

File: src/main/java/com/commercetools/sync/sdk2/products/helpers/PriceReferenceResolver.java [hunk 8]
```diff
@@ -0,0 +1,296 @@
...
+   * class. Then it resolves the supply channel reference on the supplied {@code
+   * inventoryEntryDraft} by setting the id of it's supply channel reference with the newly created
+   * Channel.
+   *
+   * <p>If the {@code ensureChannels} options is set to {@code false} on the {@code options}
+   * instance of {@code this} class, the future is completed exceptionally with a {@link
+   * ReferenceResolutionException}.
+   *
+   * <p>The method then returns a CompletionStage with a resolved channel reference {@link
+   * io.sphere.sdk.products.PriceDraftBuilder} object.
```
[the remark this answers is not in the document]
Comment (lojzatran) [reply src/main/java/com/commercetools/sync/sdk2/products/helpers/PriceReferenceResolver.java]: https://github.com/commercetools/commercetools-sync-java/pull/931/commits/37c82ff9880900242f304c60a89bf06e4ede69fd

File: src/main/java/com/commercetools/sync/sdk2/products/helpers/PriceReferenceResolver.java (hunk 4 above)
Comment (lojzatran) [reply src/main/java/com/commercetools/sync/sdk2/products/helpers/PriceReferenceResolver.java]: https://github.com/commercetools/commercetools-sync-java/pull/931/commits/37c82ff9880900242f304c60a89bf06e4ede69fd

File: src/main/java/com/commercetools/sync/sdk2/products/helpers/PriceReferenceResolver.java (hunk 5 above)
Comment (lojzatran) [reply src/main/java/com/commercetools/sync/sdk2/products/helpers/PriceReferenceResolver.java]: https://github.com/commercetools/commercetools-sync-java/pull/931/commits/53fb759945b160f8e8ad9083e7f56216bda984e8

File: src/test/java/com/commercetools/sync/sdk2/inventories/InventorySyncMockUtils.java [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+   * @return mock instance of {@link io.sphere.sdk.channels.Channel}
+   */
+  public static Channel getMockSupplyChannel(final String id, final String key) {
+    final Channel channel = mock(Channel.class);
+    when(channel.getId()).thenReturn(id);
+    when(channel.getKey()).thenReturn(key);
+    when(channel.getRoles()).thenReturn(singletonList(ChannelRoleEnum.INVENTORY_SUPPLY));
+    when(channel.toReference()).thenReturn(ChannelReferenceBuilder.of().id(id).build());
+    when(channel.toResourceIdentifier())
+        .thenReturn(ChannelResourceIdentifierBuilder.of().id(id).key(key).build());
```
[the remark this answers is not in the document]
Comment (lojzatran) [reply src/test/java/com/commercetools/sync/sdk2/inventories/InventorySyncMockUtils.java]: https://github.com/commercetools/commercetools-sync-java/pull/931/commits/f55e29673ac6011d75e5418cc1545113bfcdee99

File: src/test/java/com/commercetools/sync/sdk2/inventories/InventorySyncMockUtils.java (hunk 6 above)
Comment (lojzatran) [reply src/test/java/com/commercetools/sync/sdk2/inventories/InventorySyncMockUtils.java]: https://github.com/commercetools/commercetools-sync-java/commit/37c82ff9880900242f304c60a89bf06e4ede69fd

**Review by salander85 — APPROVED** [review]

**Pull request merged**